pkg/k8s/k8s_namespace.go

```diff
@@ -47,7 +46,7 @@ func (c *Client) GetNamespace(ctx context.Context, name string) (*corev1.Namespa
 func (c *Client) NamespaceExists(ctx context.Context, name string) bool {
 	_, err := c.GetNamespace(ctx, name)
 	if err != nil {
-		logrus.Debugf("Namespace %s does not exist, err: %v", name, err)
+		c.logger.Debugf("Namespace %s does not exist, err: %v", name, err)
```

Comment:
Similar comment, might as well make this `Errorf`

Reply:
This is usually not an error, it just tells that if the namespace exist or not.  Although we need to check if the type of the returned error is `NotFound`.
will fix it